### ISSUE TITLE
Any (synced) block patterns in a post stops blocks deeply nested in `templateLock: contentOnly` parents from being editable

### ISSUE THREAD

**Issue by wongjn** [WordPress/gutenberg#54561]
### Description

If I have a structure like:

```html
<!-- wp:group {"templateLock":"contentOnly","layout":{"type":"default"}} -->
<div class="wp-block-group">
  <!-- wp:group {"layout":{"type":"constrained"}} -->
  <div class="wp-block-group">
    <!-- wp:paragraph -->
    <p>Some text</p>
    <!-- /wp:paragraph -->
  </div>
  <!-- /wp:group -->
</div>
<!-- /wp:group -->
```

The paragraph block's text is editable.

However, when adding a synced block pattern to the content, the paragraph block is no longer editable – the immediate parent `core/group` block has `inert="true"` and `is-editing-disabled` HTML class applied to it. I have only tested with synced block patterns. It could well be the same for unsynced.

### Step-by-step reproduction instructions

1. Create a synced block pattern in the site editor with any content.
2. Create a new page.
3. Switch to the code editor, paste in the code mentioned previously.
4. Switch back to visual editor and notice that the paragraph content can be edited.
5. Add the synced block pattern from step 1 into the page content.
6. Try to edit the paragraph content now and observe how it cannot be edited.

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/11310624/1ad35a86-3ef4-4593-8bd5-6dc065111822

### Environment info

Replicated in:

- WordPress 6.3.1
- WordPress Playground 6.3
- Chrome Version 116.0.5845.188 (Official Build) (64-bit)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by Mamaduka** [comment]
I think synced patterns don't cause the issue, but it's more apparent when testing with them.

I can also reproduce the bug by adding a new paragraph after the "content locked" group (see screencast below), but the issue disappears when I save the post and refresh the page.

I will assign this to myself and continue looking for a source of the bug.

## Screencast
https://github.com/WordPress/gutenberg/assets/240569/346e68fe-2e8e-42dd-a536-e281e76db8aa

**Comment by Mamaduka** [comment]
Here's to PR to fix the issue #54618.
